src/components/spreadsheet/custom-columns/custom-columns-save.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { FormattedMessage, useIntl } from 'react-intl';
+import { IconButton } from '@mui/material';
+import SaveIcon from '@mui/icons-material/Save';
+import { ElementType, useSnackMessage, useStateBoolean } from '@gridsuite/commons-ui';
+import ElementCreationDialog, { IElementCreationDialog } from '../../dialogs/element-creation-dialog';
+import { useMemo } from 'react';
+import { createSpreadsheetModel } from '../../../services/explore';
+import { useSelector } from 'react-redux';
+import { AppState } from '../../../redux/reducer';
+import { TABLES_DEFINITION_INDEXES, TABLES_NAMES } from '../utils/config-tables';
+import { EQUIPMENT_TYPES } from '../../utils/equipment-types';
+import { SpreadsheetConfig } from '../../../types/custom-columns.types';
+
+export type CustomColumnsSaveProps = {
+    indexTab: number;
+};
+
+export default function CustomColumnsSave({ indexTab }: Readonly<CustomColumnsSaveProps>) {
+    const { snackInfo, snackError } = useSnackMessage();
+    const intl = useIntl();
+
+    const customColumnsDefinitions = useSelector(
+        (state: AppState) => state.allCustomColumnsDefinitions[TABLES_NAMES[indexTab]].columns
+    );
+    const allReorderedTableDefinitionIndexes = useSelector(
+        (state: AppState) => state.allReorderedTableDefinitionIndexes
+    );
+    const dialogOpen = useStateBoolean(false);
+
+    const currentType = useMemo(() => {
+        const equipment: any = TABLES_DEFINITION_INDEXES.get(indexTab);
```

Comment:
remove any, the type is inferred

Reply:
yes. Removing any, I saw a typing huge mistake !  (ElementType vs EQUIPMENT_TYPES)

Reply:
yes, it will be resolved by taking in consideration the last comment